homeassistant/components/saj/config_flow.py

```diff
@@ -0,0 +1,125 @@
+            try:
+                inverter = SAJInverter(user_input)
+                await inverter.connect()
+                await self.async_set_unique_id(inverter.serialnumber)
+                self._abort_if_unique_id_configured()
+                user_input[ENABLED_SENSORS] = inverter.get_enabled_sensors()
+                user_input[CONF_DEVICE_ID] = inverter.serialnumber
+
+                return self.async_create_entry(title=inverter.name, data=user_input)
+            except pysaj.UnauthorizedException:
+                errors["base"] = "invalid_auth"
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except data_entry_flow.FlowError as error:
+                raise error
+            except Exception as error:  # pylint: disable=broad-except
+                _LOGGER.error("Unexpected exception: %s", error)
+                errors["base"] = "unknown"
+        else:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+                    vol.Required(CONF_TYPE, default=user_input.get(CONF_TYPE)): vol.In(
+                        INVERTER_TYPES
+                    ),
+                    vol.Optional(CONF_NAME, default=user_input.get(CONF_NAME, "")): str,
+                    vol.Optional(
+                        CONF_USERNAME,
+                        "credentials",
+                        default=user_input.get(CONF_USERNAME, ""),
+                    ): str,
+                    vol.Optional(
+                        CONF_PASSWORD,
+                        "credentials",
+                        default=user_input.get(CONF_PASSWORD, ""),
+                    ): str,
+                }
+            ),
+            errors=errors,
+        )
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage options."""
+
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=(
+                vol.Schema(
+                    {
+                        vol.Optional(
+                            CONF_SCAN_INTERVAL,
+                            default=self.config_entry.options.get(
+                                CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                            ),
+                        ): vol.All(
+                            vol.Coerce(int),
+                            vol.Clamp(min=MIN_SCAN_INTERVAL, max=MAX_SCAN_INTERVAL),
+                        ),
+                    }
```

Comment:
We no longer need a scan interval to be configurable.
Please set a sane default, user can manually disable polling and update using automations on their own schedule. This is a generic feature of Home Assistant since 2021.6.

Reply:
30s is OK?